### PR TITLE
Add stubs to classes in zipfile to better enable subclassing

### PR DESCRIPTION
**Pull request by AlexTalks**
This adds `fp`, `NameToInfo`, and `_writecheck` to the type annotations of `zipfile.ZipFile`, as well as `FileHeader` to the annotations of `zipfile.ZipInfo`.

### CODE REVIEW

File: stdlib/2and3/zipfile.pyi [hunk 1]
```diff
@@ -48,6 +50,7 @@ class ZipFile:
     def setpassword(self, pwd: bytes) -> None: ...
     def read(self, name: _SZI, pwd: Optional[bytes] = ...) -> bytes: ...
     def testzip(self) -> Optional[str]: ...
+    def _writecheck(self, zinfo: ZipInfo) -> None: ...
```
Comment (msullivan) [inline stdlib/2and3/zipfile.pyi]: I think that typically we don't include private methods like this in typeshed stubs.

Otherwise looks good.

Comment (AlexTalks) [reply stdlib/2and3/zipfile.pyi]: Meaning that if you _were_ going to use this method, it's better to have a `# type: ignore`?

Comment (JelleZijlstra) [reply stdlib/2and3/zipfile.pyi]: Yes. Private methods aren't stable and can change without notice, so using one could well break your code on a different Python version.

File: stdlib/2and3/zipfile.pyi [hunk 2]
```diff
@@ -91,6 +94,7 @@ class ZipInfo:
                  date_time: Optional[_DT] = ...) -> None: ...
     if sys.version_info >= (3, 6):
         def is_dir(self) -> bool: ...
+    def FileHeader(self, zip64: Optional[bool] = ...) -> str: ...
```
Comment (JelleZijlstra) [inline stdlib/2and3/zipfile.pyi]: I think this returns bytes.

**Comment by AlexTalks** [comment]
@JelleZijlstra wanted to bump this - I've updated the PR based on the changes requested, so let me know if it looks okay!

**Pull request merged**